data_steward/cdr_cleaner/cleaning_rules/base_cleaning_rule.py

```diff
@@ -423,13 +424,14 @@ def affected_tables(self, affected_tables):
             self._affected_tables = []
 
     @table_namer.setter
-    def table_namer(self, table_namer, **kwargs):
+    def table_namer(self, table_namer):
         """
         Set the table_namer for this class instance. If no value is provided, it is set to a default value.
         """
         if not table_namer:
-            for key, value in kwargs.items():
-                self._table_namer = kwargs.get(value)
+            self._table_namer = re.sub('\d\d\d\dq\dr\d', '', self.dataset_id)
```

Comment:
nit: regex can be contracted 
```suggestion
            self._table_namer = re.sub('\d{4}q\dr\d', '', self.dataset_id)
```